subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/configurations/renderer/ConsoleConfigurationReportRenderer.java

```diff
@@ -0,0 +1,452 @@
+        boolean hasVariants = configs.stream().anyMatch(ReportConfiguration::hasVariants);
+
+        output.style(StyledTextOutput.Style.Info);
+        if (hasLegacy) {
+            output.println("(l) Legacy or deprecated configuration. Those are variants created for backwards compatibility which are both resolvable and consumable.");
+        }
+        if (hasIncubating) {
+            output.text("(i) Configuration uses incubating attributes such as ");
+            output.withStyle(StyledTextOutput.Style.Identifier).text("Category.VERIFICATION");
+            output.println(".");
+        }
+        if (recursiveExtensionsPrinted) {
+            output.println("(t) Configuration extended transitively.");
+        }
+        if (hasVariants) {
+            output.text("(*) Secondary variants are variants created via the ");
+            output.withStyle(StyledTextOutput.Style.Identifier).text("Configuration#getOutgoing(): ConfigurationPublications");
+            output.println(" API which also participate in selection, in addition to the configuration itself.");
+        }
+    }
+
+    private void writeConfiguration(AbstractConfigurationReportSpec spec, ReportConfiguration config) {
+        writeConfigurationNameHeader(config, spec.getReportedTypeAlias());
+        writeDescription(config.getDescription());
+
+        if (!config.getAttributes().isEmpty() ||
+            (spec.isIncludeCapabilities() && !config.getCapabilities().isEmpty()) ||
+            (spec.isIncludeArtifacts() && !config.getArtifacts().isEmpty()) ||
+            (spec.isIncludeExtensions() && !config.getExtendedConfigurations().isEmpty()) ||
+            (spec.isIncludeVariants() && !config.getSecondaryVariants().isEmpty())) {
+            newLine();
+        }
+
+        if (spec.isIncludeCapabilities()) {
+            writeCapabilities(config.getCapabilities());
+        }
+
+        writeAttributes(config.getAttributes());
+
+        if (spec.isIncludeArtifacts()) {
+            writeArtifacts(config.getArtifacts());
+        }
+        if (spec.isIncludeExtensions()) {
+            writeExtensions(config.getExtendedConfigurations(), spec.isIncludeExtensionsRecursively());
+        }
+
+        if (spec.isIncludeVariants()) {
+            writeSecondaryVariants(config.getSecondaryVariants());
+        }
+
+        newLine();
+    }
+
+    private void writeConfigurationNameHeader(ReportConfiguration config, String targetName) {
+        printHeader(() -> {
+            output.style(StyledTextOutput.Style.Normal).text(StringUtils.capitalize(targetName) + " ");
+            output.style(StyledTextOutput.Style.Header).text(config.getName());
+            output.style(StyledTextOutput.Style.Info).println(buildIndicators(config));
+        });
+    }
+
+    private void writeDescription(String description) {
+        indent(false);
+        if (description != null) {
+            output.style(StyledTextOutput.Style.Description).text("Description");
```

Comment:
I'm not sure this `Description =` is necessary. I think it looks fine if you just print the description on its own.

Reply:
I could go either way. I think the `=` looks a little weird and is inconsistent with what we show in `dependencies`.

Reply:
Ok, I'll remove it to be consistent.